### DISCUSSION TITLE
Object partial becomes unavailable if used after refine

### DISCUSSION THREAD

**Discussion by darklight9811** [colinhacks/zod#1802] [Q&A] [answered]
I don't know if it was intentional, but I actually don't see the reason for this to be done.

**Comment by JacobWeisenburger** [comment, marked as answer]
`refine` returns `ZodEffects` which is not the same as `ZodObject`.

You can use `innerType` to get the `ZodObject` and call `.partial` on that.
```ts
z.object( {} ).refine( () => { } ).innerType().partial()
```
